docker-entrypoint.sh

```diff
@@ -11,15 +11,18 @@ if [[ "$1" == apache2* ]] || [ "$1" == php-fpm ]; then
 		# if not specified, let's generate a random value
 		: "${YOURLS_COOKIEKEY:=$(head -c1m /dev/urandom | sha1sum | cut -d' ' -f1)}"
 
-		if [ ! -e /var/www/html/user/config.php ]; then
+		# We want to copy the initial config if the actual config file doesn't already
+		# exist OR if it is an empty file (e.g. it has been created for the volume mount).
+		if [ ! -e /var/www/html/user/config.php ] || [ ! -s /var/www/html/user/config.php ]; then
 			cp /var/www/html/user/config-docker.php /var/www/html/user/config.php
 			chown www-data:www-data /var/www/html/user/config.php
 		fi
 
 		: "${YOURLS_USER:=}"
 		: "${YOURLS_PASS:=}"
 		if [ -n "${YOURLS_USER}" ] && [ -n "${YOURLS_PASS}" ]; then
-			sed -i "s/  getenv('YOURLS_USER') => getenv('YOURLS_PASS'),/  '${YOURLS_USER}' => '${YOURLS_PASS}',/g" /var/www/html/user/config.php
+			result=$(sed "s/  getenv('YOURLS_USER') => getenv('YOURLS_PASS'),/  \'${YOURLS_USER}\' => \'${YOURLS_PASS}\',/g" /var/www/html/user/config.php)
+			echo "$result" > /var/www/html/user/config.php
```

Comment:
I don't understand this change... Could explain a bit for me? Thx 🙂

Reply:
Sure. The problem this fixes is that when you use `sed -i`, it creates a temporary file then moves that to replace the original file. That won't work if the file has been mounted because Docker mounts it as a filing system (see `df -a` within a container when you do this).

So, instead, the sed command is run without `-i` and the results stored in a variable. Those results are then written back out to the config file.

You could, in theory, combine the two lines but shellcheck advises against doing it all in one pipeline, hence the two separate steps.

Reply:
Thanks!